src/App.tsx

```diff
@@ -2,15 +2,24 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { useState } from 'react';
 
 export const App = () => {
+  const [movies, setMovies] = useState(moviesFromServer);
+
   return (
     <div className="page">
       <div className="page-content">
-        <MoviesList movies={moviesFromServer} />
+        <MoviesList movies={movies} />
       </div>
       <div className="sidebar">
-        <NewMovie /* onAdd={(movie) => {}} */ />
+        <NewMovie
+          onAdd={movie => {
+            const newMovies = [...movies, movie];
+
+            setMovies(newMovies);
```

Comment:
```suggestion
            setMovies(prevMovies => [...prevMovies, movie]);
```

https://dev.to/gakii/functional-state-update-in-react-42io

Reply:
> https://dev.to/gakii/functional-state-update-in-react-42io

Not fixed